chia/cmds/db_validate_func.py

```diff
@@ -102,8 +103,15 @@ def validate_v2(in_path: Path, *, validate_blocks: bool) -> None:
                     continue
 
                 if validate_blocks:
-                    block = FullBlock.from_bytes(zstd.decompress(row[4]))
-                    block_record = BlockRecord.from_bytes(row[5])
+                    block_bytes = zstd.decompress(row[4])
+                    block = FullBlock.from_bytes(block_bytes)
+                    block_record_db: BlockRecordDB = BlockRecordDB.from_bytes(row[5])
+                    plot_filter_info: PlotFilterInfo = plot_filter_info_from_block(block_bytes)
+                    block_record = BlockRecord.from_block_record_db(
+                        block_record_db,
+                        plot_filter_info.pos_ss_cc_challenge_hash,
+                        plot_filter_info.cc_sp_hash,
+                    )
```

Comment:
renaming `block_record` to `block_record_db` and constructing this `block_record` object seems unnecessary, and just increases complexity. Unless I'm missing something, I would suggest you preserve the name of `block_record` (but obviously you need to change its type to `BlockRecordDB`.

In the code below where you check the `pos_cc_challenge_hash` and `pos_ss_cc_challenge_hash` you can just compare them against `plot_filter_info`.

But, come to think of it, all of this new code seems unnecessary. This is not validating the integrity of the database. You effectively parse out these fields using two separate functions and then ensure they are the same. It looks like a poor-man's test of your new parser function. I don't think that belongs in here. Add it to the unit test instead

Reply:
+1